app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -159,7 +163,13 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
                         .edit()
                         .putInt(SHOW_UNPIN_TOAST_COUNTER_PREF, prefInt + 1)
                         .apply()
-                Toast.makeText(context, R.string.homescreen_unpin_tutorial_toast, Toast.LENGTH_LONG).show()
+
+                // With VoiceView enabled, the toast is not announced if it disappears before the home tile description
+                // is finished. This adds a short delay to the display of the toast so it will always be announced
+                if (context.isVoiceViewEnabled())
+                    uiHandler.postDelayed(
+                        { Toast.makeText(context, R.string.homescreen_unpin_tutorial_toast, Toast.LENGTH_LONG).show() }, 1500)
```

Comment:
We want to avoid capturing context in this lambda, since it could potentially outlive the fragment.  That would briefly leak the fragment, and I'm not sure what happens when we call `makeText` with a destroyed context but I'm sure it's not good.

Suggest wrapping the context in a WeakReference, then unwrapping it in the lambda.  Make sure to set up the WeakReference before the call to `postDelayed`, or the lambda will implicitly capture the context anyway.